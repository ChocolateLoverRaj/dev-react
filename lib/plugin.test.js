/* global test, expect */

import createPlugin from './plugin.js'

import babel from '@babel/core'

test('transform', async () => {
  const inputCode = `
import React from 'react'
import { a, b } from 'something'
import * as m from 'stuff'
import keepDoubleQuotes from "double-quotes"
import keepSingleQuotes from 'single-quotes'

import DoNotChange from 'unchanged'
  `
  const { code } = await babel.transformAsync(inputCode, {
    plugins: [createPlugin(new Map()
      .set('react', '/scripts/react.js')
      .set('something', 'something-else')
      .set('stuff', 'some-stuff')
      .set('double-quotes', 'd-quotes')
      .set('single-quotes', 's-quotes')
    )]
  })
  expect(code).toMatchSnapshot()
})
