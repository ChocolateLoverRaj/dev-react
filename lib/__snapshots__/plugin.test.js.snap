// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform 1`] = `
"import React from '/scripts/react.js';
import { a, b } from 'something-else';
import * as m from 'some-stuff';
import keepDoubleQuotes from \\"d-quotes\\";
import keepSingleQuotes from 's-quotes';
import DoNotChange from 'unchanged';"
`;
